indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/PendingTaskBasedWorkerProvisioningStrategy.java

```diff
@@ -121,11 +151,12 @@ private PendingProvisioner(WorkerTaskRunner runner)
     public synchronized boolean doProvision()
     {
       Collection<Task> pendingTasks = runner.getPendingTaskPayloads();
+      log.debug("Pending tasks: %d %s", pendingTasks.size(), pendingTasks);
       Collection<ImmutableWorkerInfo> workers = runner.getWorkers();
+      log.debug("Workers: %d %s", workers.size(), workers);
       boolean didProvision = false;
-      final WorkerBehaviorConfig workerConfig = workerConfigRef.get();
-      if (workerConfig == null || workerConfig.getAutoScaler() == null) {
-        log.error("No workerConfig available, cannot provision new workers.");
+      final WorkerBehaviorConfig workerConfig = getWorkerBehaviorConfig(workerConfigRef, "provision", log);
+      if (workerConfig == null) {
```

Comment:
Is the eager checking for null autoscaler no longer needed?

Reply:
This is done inside `getWorkerBehaviorConfig()` now